### DISCUSSION TITLE
Camera Rotation

### DISCUSSION THREAD

**Discussion by BrunoXL** [microsoft/AirSim#4004] [Support Q&A]
I'm using `ComputerVision` mode and I would like to rotate AirSim's front camera.  
For that, I followed the instruction on #222 about rotate the camera, and it worked fine.  
However, doing that messy the keyboard keys for moving the drone (e.g. left key doesn't move camera to the left, but moves to forward instead).
My question is: how can I rotate 'only' the camera without this side effect. 

EDIT:
It seems there is a way to make it on `settings.json`

Below, follows my configuration.

      {
          "SeeDocsAt": "https://github.com/Microsoft/AirSim/blob/master/docs/settings.md",
          "SettingsVersion": 1.2,
          "SimMode": "ComputerVision",
          "ViewMode": "Manual",
          "Vehicles": {
          	"SimpleFlight": {
          		"VehicleType":"SimpleFlight",
          		"DefaultVehicleName": "Armed",
          		"AutoCreate": true,
          		"PawnPaths": "",
          		"EnableCollisionsPassthrough": false,
          		"EnableCollisions": true,
          		"AllowAPIAlways": true,
          		"RC":{
          			"RemoteControlID": 0,
          			"AllowAPIWhenDisconnectd": false
          		  }
          	},
          	"Retrieve_Images": {
          		"VehicleType":"ComputerVision",
          		"DefaultVehicleName": "Armed",
          		"AutoCreate": true,
          		"PawnPaths": "",
          		"EnableCollisionsPassthrough": false,
          		"EnableCollisions": true,
          		"AllowAPIAlways": true,
          		"RC":{
          			"RemoteControlID": 0,
          			"AllowAPIWhenDisconnectd": false
          	  	},
              "Cameras":{
                  "Test":{
                    "CaptureSettings": [
                      {
                      "ImageType": 0,
                      "Width": 1920,
                      "Height": 1080,
                      "FOV_Degrees": 90,
                      "AutoExposureSpeed": 100,
                      "AutoExposureBias": 0,
                      "AutoExposureMaxBrightness": 0.64,
                      "AutoExposureMinBrightness": 0.03,
                      "MotionBlurAmount": 0,
                      "TargetGamma": 1.0,
                      "ProjectionMode": "",
                      "OrthoWidth": 5.12
                    }  
                  ],
                  "Roll": 90, "Pitch": 90, "Yaw": 90
                }
              }
            }
          }  
        }

This files thrown this error:

     [2018.10.24-20.26.11:972][664]LogActor: Warning: BP_CameraDirector_C 
     /Game/UEDPIE_0_Rodain.Rodain:PersistentLevel.CameraDirector has natively added scene 
    component(s), but none of them were set as the actor's RootComponent - picking one arbitrarily
    [2018.10.24-20.26.11:984][664]LogPawn: Error: EnableInput can only be specified on a Pawn for its 
    Controller
    [2018.10.24-20.26.11:984][664]LogSpawn: Warning: SpawnActor failed because the given transform 
    (nan,nan,nan|90.000000,-13.262693,-13.262695|1.000000,1.000000,1.000000) is invalid
    [2018.10.24-20.26.12:038][664]LogLinux: === Critical error: ===
    Unhandled Exception: SIGSEGV: invalid attempt to read memory at address 0x0000000000000178

    [2018.10.24-20.26.12:038][664]LogLinux: Fatal error!

**Comment by dasmehdix** [comment]
You can change camera rotation within UE4 editor. After open your project, check your plugins and enter "/Airsim/blueprint/BP_Flyingpawn". At the left you can see "FrontCamera". Just change the rotation values of that object.
